ignite/metrics/metric.py

```diff
@@ -148,7 +155,7 @@ def completed(self, engine: Engine, name: str) -> None:
 
             engine.state.metrics[name] = result
 
-    def attach(self, engine: Engine, name: str) -> None:
+    def attach(self, engine: Engine, name: Optional[str] = None) -> None:
```

Comment:
Related to comment above. IMO name shouldn’t be None.

Reply:
This was to maintain compatibility with the way `MetricsLambda` attaches only to `EPOCH_COMPLETED` and checks for partially attached metric objects. It checks that internal metric objects are attached to [particular events only](https://github.com/pytorch/ignite/blob/95d4fe67029d7c78567ce8969b4a7e6c48701b9f/ignite/metrics/metrics_lambda.py#L127). Only the top level object should attach to `EPOCH_COMPLETED` and all others [attached partially](https://github.com/pytorch/ignite/blob/95d4fe67029d7c78567ce8969b4a7e6c48701b9f/ignite/metrics/metrics_lambda.py#L100). This was all based on my understanding of how this mechanism worked.